### PR TITLE
Add optional OLM build checks

### PR DESCRIPTION
**Pull request by timflannagan**
Update the downstream OLM prow configuration and add image build steps for the OLM and operator-registry container images.

Make these checks optional for now as we expect these to fail in the current repository layout.

### CODE REVIEW

File: ci-operator/config/openshift/operator-framework-olm/openshift-operator-framework-olm-master.yaml [hunk 1]
```diff
@@ -1,7 +1,33 @@
+base_images:
+  base:
+    name: "4.8"
+    namespace: ocp
+    tag: base
+  ocp_builder_rhel-8-golang-1.15-openshift-4.8:
+    name: builder
+    namespace: ocp
+    tag: rhel-8-golang-openshift-4.8
```
Comment (timflannagan) [inline ci-operator/config/openshift/operator-framework-olm/openshift-operator-framework-olm-master.yaml:9]: This is what's currently in the root OLM/registry Dockerfile(s) for the tag being specified. We'll likely need to update this to use the more explicit rhel-8-golang-1.15-openshift-4.8 at some point.

Comment (timflannagan) [reply ci-operator/config/openshift/operator-framework-olm/openshift-operator-framework-olm-master.yaml]: Ah, nevermind, there's a mismatch between what base image tags we specify in the registry vs. OLM Dockerfile(s). Should be fine now.

File: ci-operator/config/openshift/operator-framework-olm/openshift-operator-framework-olm-master.yaml [hunk 2]
```diff
@@ -11,10 +40,31 @@ tag_specification:
...
 tests:
-- as: unit
-  commands: ls /usr/local/kubebuilder/bin
+- as: verify
+  commands: |
+    make vendor && git diff --stat HEAD --ignore-submodules --exit-code
+  container:
+    from: src
+- as: unit-olm
+  commands: go env && make unit/olm
```
Comment (timflannagan) [inline ci-operator/config/openshift/operator-framework-olm/openshift-operator-framework-olm-master.yaml:49]: Hmm, this appears to be problematic as we removed the nested vendor directory from VCS in the downstream repository so running any `go test ...` arguments attempt to build that tree as we go, but there are conflicts present in the modules defined.

**Comment by timflannagan** [comment]
Made some updates to the unit helper script.

/retest

**Comment by timflannagan** [comment]
Hmm, I don't think those changes propagated.

/test all

**Review by ankitathomas — APPROVED** [review]
/lgtm

**Pull request merged**